src/xcode/ENA/ENA/Source/Scenes/Events/OnBehalfCheckinSubmission/TraceLocationSelection/OnBehalfTraceLocationSelectionViewController.swift

```diff
@@ -0,0 +1,246 @@
+////
+// 🦠 Corona-Warn-App
+//
+
+import UIKit
+import OpenCombine
+
+class OnBehalfTraceLocationSelectionViewController: UITableViewController, DismissHandling, FooterViewHandling {
+
+	// MARK: - Init
+	
+	init(
+		viewModel: OnBehalfTraceLocationSelectionViewModel,
+		onScanQRCodeCellTap: @escaping () -> Void,
+		onMissingPermissionsButtonTap: @escaping () -> Void,
+		onPrimaryButtonTap: @escaping (TraceLocation) -> Void,
+		onDismiss: @escaping () -> Void
+	) {
+		self.viewModel = viewModel
+		self.onScanQRCodeCellTap = onScanQRCodeCellTap
+		self.onMissingPermissionsButtonTap = onMissingPermissionsButtonTap
+		self.onPrimaryButtonTap = onPrimaryButtonTap
+		self.onDismiss = onDismiss
+		
+		super.init(style: .plain)
+	}
+	
+	required init?(coder: NSCoder) {
+		fatalError("init(coder:) has not been implemented")
+	}
+	
+	// MARK: - Overrides
+	
+	override func viewDidLoad() {
+		super.viewDidLoad()
+
+		parent?.title = viewModel.title
+		parent?.navigationItem.largeTitleDisplayMode = .always
+		parent?.navigationItem.rightBarButtonItem = dismissHandlingCloseBarButton
+
+		setUpTableView()
+
+		tableView.reloadData()
+		setUpEmptyState()
+		
+		viewModel.$continueEnabled
+			.receive(on: DispatchQueue.main.ocombine)
+			.sink { [weak self] in
+				self?.footerView?.setEnabled($0, button: .primary)
+			}
+			.store(in: &subscriptions)
+
+		viewModel.$triggerReload
+			.receive(on: DispatchQueue.main.ocombine)
+			.sink { [weak self] _ in
+				guard let self = self else { return }
+
+				self.tableView.reloadData()
+				self.setUpEmptyState()
+			}
+			.store(in: &subscriptions)
+	}
+
+	override func viewWillAppear(_ animated: Bool) {
+		super.viewWillAppear(animated)
+
+		navigationController?.navigationBar.prefersLargeTitles = true
+		navigationController?.navigationBar.sizeToFit()
```

Comment:
This call looks strange to me in times of auto layout. Is this needed to fix some strange layout behavior?

Reply:
Yes, we use it in some other places to force the large title to show immediately. But I removed it now since it was not working here anyway 🤷🏼‍♂️